src/algokit_client_generator/generator.py

```diff
@@ -349,8 +367,15 @@ def __init__(self, app_client: algokit_utils.ApplicationClient, atc: AtomicTrans
     def build(self) -> AtomicTransactionComposer:
         return self.atc
 
-    def simulate(self) -> SimulateAtomicTransactionResponse:
-        result = self.atc.simulate(self.app_client.algod_client)
+    def simulate(self, options: typing.Optional[SimulateOptions] = None) -> SimulateAtomicTransactionResponse:
```

Comment:
Can use `options: SimulateOptions | None = None` here. Also could maybe just use `models.SimulateRequest` since we are already exposing other algosdk types, this would mean don't need to keep SimulateOptions in sync with the sdk?

Reply:
I actually had the union type but swapped to `Optional` as it seems to be used elsewhere (in the sdk)

The reason for duplicating the class is that `models.SimulateRequest` asks for an array of transactions which `atc.simulate` promptly ignores. Rather than putting that awkward API on the consumer it seemed preferable to hide it away. In the TS version we could do some magical conditional types to extract properties from the `SimulateRequest` type but no such luck here

Reply:
The SDK is supporting earlier versions of python where the `| None` style wasn't supported, but our generated client is already using `| None` as we target 3.10+

Reply:
ah ok, makes sense. I'll swap to ` | None`